From 64ef7525a86e9ed9acc568c753e7637820d57aea Mon Sep 17 00:00:00 2001
From: Keyon Jie <yang.jie@linux.intel.com>
Date: Tue, 22 May 2018 18:23:43 +0800
Subject: [PATCH 1658/2313] ASoC: Intel: Boards: Add BXTP MRB machine driver
 for NXP TDF8532

This is the machine driver for NXP TDF8532, from production
kernel, rebased it to sof-v4.14 base.

Signed-off-by: Sinha, Mohit <mohit.sinha@intel.com>
Signed-off-by: Markus Schweikhardt <markus.schweikhardt@intel.com>
Signed-off-by: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
Signed-off-by: Kareem,Shaik <kareem.m.shaik@intel.com>
Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>
---
 sound/soc/intel/boards/Kconfig       | 9 +++++++++
 sound/soc/intel/boards/Makefile      | 2 ++
 sound/soc/intel/boards/bxt_tdf8532.c | 7 ++-----
 3 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/sound/soc/intel/boards/Kconfig b/sound/soc/intel/boards/Kconfig
index 325cf83..760fe5d 100644
--- a/sound/soc/intel/boards/Kconfig
+++ b/sound/soc/intel/boards/Kconfig
@@ -248,6 +248,15 @@ config SND_SOC_INTEL_BXT_PCM512x_MACH
 	  This adds support for ASoC machine driver for Broxton platforms
 	  with TI PCM512x I2S audio codec.
 	  Say Y or m if you have such a device. This is a recommended option.
+
+config SND_SOC_INTEL_BXT_TDF8532_MACH
+	tristate "ASoC Audio driver for BXT with TDF8532 in I2S mode"
+	depends on X86 && ACPI && I2C
+	select SND_SOC_TDF8532
+	select SND_SOC_COMPRESS
+	help
+	  This adds support for ASoC machine driver for Broxton IVI GP MRB platform
+	  Say Y if you have such a device.
 	  If unsure select "N".
 
 endif ##  SND_SOC_INTEL_SKL || SND_SOC_SOF_APOLLOLAKE
diff --git a/sound/soc/intel/boards/Makefile b/sound/soc/intel/boards/Makefile
index c21bd2f..646b293b 100644
--- a/sound/soc/intel/boards/Makefile
+++ b/sound/soc/intel/boards/Makefile
@@ -7,6 +7,7 @@ snd-soc-sst-broadwell-objs := broadwell.o
 snd-soc-sst-bxt-da7219_max98357a-objs := bxt_da7219_max98357a.o
 snd-soc-sst-bxt-rt298-objs := bxt_rt298.o
 snd-soc-sst-bxt-pcm512x-objs := bxt_pcm512x.o
+snd-soc-sst_bxt_tdf8532-objs := bxt_tdf8532.o
 snd-soc-sst-bytcr-rt5640-objs := bytcr_rt5640.o
 snd-soc-sst-bytcr-rt5651-objs := bytcr_rt5651.o
 snd-soc-sst-cht-bsw-rt5672-objs := cht_bsw_rt5672.o
@@ -36,6 +37,7 @@ obj-$(CONFIG_SND_SOC_INTEL_BYT_MAX98090_MACH) += snd-soc-sst-byt-max98090-mach.o
 obj-$(CONFIG_SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH) += snd-soc-sst-bxt-da7219_max98357a.o
 obj-$(CONFIG_SND_SOC_INTEL_BXT_RT298_MACH) += snd-soc-sst-bxt-rt298.o
 obj-$(CONFIG_SND_SOC_INTEL_BXT_PCM512x_MACH) += snd-soc-sst-bxt-pcm512x.o
+obj-$(CONFIG_SND_SOC_INTEL_BXT_TDF8532_MACH) += snd-soc-sst_bxt_tdf8532.o
 obj-$(CONFIG_SND_SOC_INTEL_BROADWELL_MACH) += snd-soc-sst-broadwell.o
 obj-$(CONFIG_SND_SOC_INTEL_BDW_RT5677_MACH) += snd-soc-sst-bdw-rt5677-mach.o
 obj-$(CONFIG_SND_SOC_INTEL_BYTCR_RT5640_MACH) += snd-soc-sst-bytcr-rt5640.o
diff --git a/sound/soc/intel/boards/bxt_tdf8532.c b/sound/soc/intel/boards/bxt_tdf8532.c
index f904e18..38dc1af 100644
--- a/sound/soc/intel/boards/bxt_tdf8532.c
+++ b/sound/soc/intel/boards/bxt_tdf8532.c
@@ -37,7 +37,6 @@ static const struct snd_soc_dapm_widget broxton_tdf8532_widgets[] = {
 };
 
 static const struct snd_soc_dapm_route broxton_tdf8532_map[] = {
-
 	/* Speaker BE connections */
 	{ "Speaker", NULL, "ssp4 Tx"},
 	{ "ssp4 Tx", NULL, "codec0_out"},
@@ -76,9 +75,8 @@ static const struct snd_soc_dapm_route broxton_tdf8532_map[] = {
 	{ "ssp3 Tx", NULL, "Modem_ssp3_out"},
 };
 
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL)
 static int bxt_tdf8532_ssp2_fixup(struct snd_soc_pcm_runtime *rtd,
-				struct snd_pcm_hw_params *params)
+				  struct snd_pcm_hw_params *params)
 {
 	struct snd_mask *fmt = hw_param_mask(params, SNDRV_PCM_HW_PARAM_FORMAT);
 
@@ -88,7 +86,6 @@ static int bxt_tdf8532_ssp2_fixup(struct snd_soc_pcm_runtime *rtd,
 
 	return 0;
 }
-#endif
 
 /* broxton digital audio interface glue - connects codec <--> CPU */
 static struct snd_soc_dai_link broxton_tdf8532_dais[] = {
@@ -382,7 +379,7 @@ static struct snd_soc_dai_link broxton_tdf8532_dais[] = {
 };
 
 static int bxt_add_dai_link(struct snd_soc_card *card,
-			struct snd_soc_dai_link *link)
+			    struct snd_soc_dai_link *link)
 {
 	link->platform_name = "0000:00:0e.0";
 	link->nonatomic = 1;
-- 
2.7.4

